File: src/main/java/com/bdev/hogwarts_api/dao/StudentStatusDao.java
package com.bdev.hogwarts_api.dao;

import com.bdev.hogwarts_api.data.model.student_status.StudentStatusModel;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface StudentStatusDao extends JpaRepository<StudentStatusModel, Long> {
    List<StudentStatusModel> findAllByStudentId(long studentId);
    StudentStatusModel findTopByStudentIdOrderByCreationTimeDesc(long studentId);
}

File: src/main/java/com/bdev/hogwarts_api/dao/StudentStatusDao.kt
package com.bdev.hogwarts_api.dao

import com.bdev.hogwarts_api.data.model.student_status.StudentStatusModel
import org.springframework.data.jpa.repository.JpaRepository

interface StudentStatusDao : JpaRepository<StudentStatusModel, Long> {
    fun findAllByStudentLogin(studentLogin: String): List<StudentStatusModel>
    fun findTopByStudentLoginOrderByCreationTimeDesc(studentLogin: String): StudentStatusModel?
}
